platform/ios/src/MGLAnnotationView.mm

```diff
@@ -212,13 +212,21 @@ - (void)setDragState:(MGLAnnotationViewDragState)dragState animated:(BOOL)animat
         [self.mapView.calloutViewForSelectedAnnotation dismissCalloutAnimated:animated];
         [self.superview bringSubviewToFront:self];
     }
-
-    if (dragState == MGLAnnotationViewDragStateEnding)
+    else if (dragState == MGLAnnotationViewDragStateCanceling)
+    {
+        self.panGestureRecognizer.enabled = NO;
+        self.longPressRecognizer.enabled = NO;
+        self.center = [self.mapView convertCoordinate:self.annotation.coordinate toPointToView:self.mapView];
+        self.panGestureRecognizer.enabled = YES;
+        self.longPressRecognizer.enabled = YES;
+        self.dragState = MGLAnnotationViewDragStateNone;
+    }
+    else if (dragState == MGLAnnotationViewDragStateEnding)
     {
-        if ([self.mapView.delegate respondsToSelector:@selector(mapView:didDragAnnotationView:toCoordinate:)])
+        if ([self.annotation respondsToSelector:@selector(setCoordinate:)])
         {
             CLLocationCoordinate2D coordinate = [self.mapView convertPoint:self.center toCoordinateFromView:self.mapView];
-            [self.mapView.delegate mapView:self.mapView didDragAnnotationView:self toCoordinate:coordinate];
+            [(NSObject *)self.annotation setValue:[NSValue valueWithMGLCoordinate:coordinate] forKey:@"coordinate"];
```

Comment:
@1ec5 I hesitated between this and NSInvocation. Wasn't sure setValue:forKey would work if the class would be Swift native base class but fortunately, the protocol ensures NSObject.